stingray/utils.py

```diff
@@ -116,6 +116,88 @@ def rebin_data(x, y, dx_new, method='sum'):
     return xbin, ybin, step_size
 
 
+def rebin_data_lc(x, y, dx_new, method='sum'):
+
+    """
+    Rebin some data to an arbitrary new data resolution. Either sum
+    the data points in the new bins or average them.
+
+    Parameters
+    ----------
+    x: iterable
+        The dependent variable with some resolution dx_old = x[1]-x[0]
+
+    y: iterable
+        The independent variable to be binned
+
+    dx_new: float
+        The new resolution of the dependent variable x
+
+    method: {"sum" | "average" | "mean"}, optional, default "sum"
+        The method to be used in binning. Either sum the samples y in
+        each new bin of x, or take the arithmetic mean.
+
+
+    Returns
+    -------
+    xbin: numpy.ndarray
+        The midpoints of the new bins in x
+
+    ybin: numpy.ndarray
+        The binned quantity y
+    """
+
+    y = np.asarray(y)
+    x = np.asarray(x)
+
+    dx_old = x[1] - x[0]
+
+    assert dx_new >= dx_old, "New frequency resolution must be larger than " \
+                             "old frequency resolution."
+
+    step_size = dx_new / dx_old
+
+    output = []
+    for i in np.arange(0, y.shape[0], step_size):
+        total = 0
+
+        int_i = int(i)
+        prev_frac = int_i + 1 - i
+        prev_bin = int_i
+        total += prev_frac * y[prev_bin]
+
+        if i + step_size < len(x):
+            # Fractional part of next bin:
+            next_frac = i + step_size - int(i + step_size)
+            next_bin = int(i + step_size)
+            total += next_frac * y[next_bin]
+
+        total += sum(y[int(i+1):int(i+step_size)])
+        output.append(total)
+
+    output = np.asarray(output)
+
+    if method in ['mean', 'avg', 'average', 'arithmetic mean']:
+        ybin = output / np.float(step_size)
+
+    elif method == "sum":
+        ybin = output
+
+    else:
+        raise UnrecognizedMethod("Method for summing or averaging not recognized. "
+                                 "Please enter either 'sum' or 'mean'.")
+
+    tseg = x[-1] - x[0] + dx_old
+
+    if (tseg / dx_new % 1) > 0:
+        ybin = ybin[:-1]
+
+    # The lightcurve time array gives the central value of the bin
+    xbin = np.arange(ybin.shape[0]) * dx_new + ((x[0]+x[int(step_size)-1])/2)
```

Comment:
I think we should change `((x[0]+x[int(step_size)-1])/2)` in accordance with odd and even values of `step_size`